### PR TITLE
Took 475 ton mapping reinhard aces

### PR DESCRIPTION
**Pull request by fcturan20**
[TOOK-475](https://oyunteknolojileri.atlassian.net/browse/TOOK-475?atlOrigin=eyJpIjoiMDIxYTQ0ZGNkZGZlNGU5OGJjZTYxMzI4YjVlNmQzYzAiLCJwIjoiaiJ9)

### CODE REVIEW

File: Editor/App.cpp [hunk 1]
```diff
@@ -196,6 +196,8 @@ namespace ToolKit
           myEditorRenderer->m_params.App      = this;
           myEditorRenderer->m_params.LitMode  = m_sceneLightingMode;
           myEditorRenderer->m_params.Viewport = viewport;
+          myEditorRenderer->m_params.AcesTonemapper =
```
Comment (afraidofdark) [inline Editor/App.cpp:199]: why aces tonemapper if it also contains reinhard implementation ?

ToneMapper looks more suitable ?

File: ToolKit/Pass.h [hunk 2]
```diff
@@ -283,10 +283,39 @@ namespace ToolKit
     ShaderPtr m_gammaShader       = nullptr;
   };
 
+  struct ToneMapPassParams
+  {
+    FramebufferPtr FrameBuffer = nullptr;
+    bool AcesTonemapper        = true; //!-< If false, Tonemapper is Reinhard
```
Comment (afraidofdark) [inline ToolKit/Pass.h:289]: use an enum TonePamTechniques { Reinhard, Aces }

File: ToolKit/Pass.h [hunk 3]
```diff
@@ -283,10 +283,39 @@ namespace ToolKit
...
+    ShaderPtr m_tonemapShader     = nullptr;
+  };
+
+  typedef std::shared_ptr<ToneMapPass> ToneMapPassPtr;
+
   struct SceneRenderPassParams
   {
     RenderPassParams renderPassParams;
     ShadowPassParams shadowPassParams;
+    int acesTonemapper = 0; // !-< 0: Tonemap off, 1: Reinhard, 2: ACES
```
Comment (afraidofdark) [inline ToolKit/Pass.h:318]: use enum

File: ToolKit/Renderer.cpp [hunk 4]
```diff
@@ -1779,6 +1779,9 @@ namespace ToolKit
         case ParameterVariant::VariantType::Int:
           glUniform1i(loc, var.second.GetVar<int>());
           break;
+        case ParameterVariant::VariantType::UInt:
```
Comment (afraidofdark) [inline ToolKit/Renderer.cpp:1782]: is this valid in opengl es 3.0 shader model 3.0 ?

**Review by afraidofdark — CHANGES_REQUESTED** [review]

File: ToolKit/Renderer.cpp (hunk 4 above)
Comment (fcturan20) [reply ToolKit/Renderer.cpp:1782]: 1uiv is valid, so it should be. We're already using it for bool.

File: Editor/App.cpp (hunk 1 above)
Comment (fcturan20) [reply Editor/App.cpp:199]: I agree that Aces is better but it's better to give user the choice. We don't have complicated tonemapper settings (max white point etc.) anyway, at least this should be available.

**Review by afraidofdark — APPROVED** [review]

**Pull request merged**